src/Uno.UI/UI/Xaml/Controls/TextBlock/TextBlock.cs

```diff
@@ -422,8 +422,12 @@ void refreshForeground()
 #endif
 			}
 
-			_foregroundChanged.Disposable =
-				Brush.AssignAndObserveBrush(Foreground, c => refreshForeground(), refreshForeground);
+			_foregroundChanged.Disposable = null;
+			if (Foreground?.SupportsAssignAndObserveBrush ?? false)
```

Comment:
@davidjohnoliver do you remember why this was not integrated inside `Brush.AssignAndObserveBrush`?

Reply:
May I guess @jeromelaban ? :smile:

There could be cases where we don't want this to silently do nothing, ie, the failure may be serving as a reminder to put any necessary `else if`s. For example:

https://github.com/unoplatform/uno/blob/33c3a26c935361432ad713c2e8debc521616c6be/src/Uno.UI/UI/Xaml/Controls/Border/BorderLayerRenderer.iOSmacOS.cs#L156-L161

Reply:
`ImageBrush` generally requires more complex handling than the `AssignAndObserveBrush()` signature allows for, IIRC, because the image may be fetched and/or processed asynchronously. That's why the exception is expected here.

The 'correct' answer is to tackle https://github.com/unoplatform/uno/issues/5192, but that's a big one.